VmAgent.Core/Interfaces/ProcessWrapper.cs

```diff
@@ -35,6 +34,24 @@ public int Start(ProcessStartInfo startInfo)
             return Process.Start(startInfo).Id;
         }
 
+        public int StartWithEventHandler(ProcessStartInfo startInfo, Action<object, DataReceivedEventArgs> StdOutputHandler, Action<object, DataReceivedEventArgs> ErrorOutputHandler, Action<object, EventArgs> ProcessExitedHanlder)
+        {
+            startInfo.RedirectStandardOutput = true;
+            startInfo.RedirectStandardError = true;
+
+            Process process = new Process();
+            process.StartInfo = startInfo;
+            process.OutputDataReceived += new DataReceivedEventHandler(StdOutputHandler);
+            process.ErrorDataReceived += new DataReceivedEventHandler(ErrorOutputHandler);
+            process.Exited += new EventHandler(ProcessExitedHanlder);
```

Comment:
ProcessExitedHanlder -> ProcessExitedHandler (this typo appears in several places :p )